app/models/piece.rb

```diff
@@ -16,4 +16,65 @@ class Piece < ActiveRecord::Base
   def types
     %w(Pawn Rook Knight Bishop Queen King)
   end
+
+  def obstructed?(destination_x, destination_y)
+    x_difference = x_position - destination_x
+    y_difference = y_position - destination_y
+    return true if diagonal?(destination_x, destination_y, x_difference, y_difference)
```

Comment:
For these method names, I'd do something like `diagonal_obstruction?`

Otherwise, if someone just reads `diagonal`, they might not be sure what it's for